packages/server/modules/workspaces/tests/integration/roles.graph.spec.ts

```diff
@@ -0,0 +1,179 @@
+import { AllScopes } from '@/modules/core/helpers/mainConstants'
+import { mapMainWorkspaceRoleToGqlRole } from '@/modules/workspaces/helpers/roles'
+import {
+  BasicTestWorkspace,
+  createTestWorkspace
+} from '@/modules/workspaces/tests/helpers/creation'
+import {
+  BasicTestUser,
+  createAuthTokenForUser,
+  createTestUser
+} from '@/test/authHelper'
+import {
+  GetWorkspaceDocument,
+  UpdateWorkspaceRoleDocument
+} from '@/test/graphql/generated/graphql'
+import {
+  createTestContext,
+  testApolloServer,
+  TestApolloServer
+} from '@/test/graphqlHelper'
+import { beforeEachContext } from '@/test/hooks'
+import { Roles } from '@speckle/shared'
+import { expect } from 'chai'
+
+describe('Workspaces Roles GQL', () => {
+  let apollo: TestApolloServer
+  let workspace: BasicTestWorkspace
+
+  const testAdminUser: BasicTestUser = {
+    id: '',
+    name: 'John Speckle',
+    email: 'john-speckle-workspace-admin@example.org',
+    role: Roles.Server.Admin
+  }
+
+  const testMemberUser: BasicTestUser = {
+    id: '',
+    name: 'James Speckle',
+    email: 'james-speckle-workspace-member@example.org',
+    role: Roles.Server.User
+  }
+
+  before(async () => {
+    await beforeEachContext()
+    await Promise.all(
+      [testAdminUser, testMemberUser].map((user) => createTestUser(user))
+    )
+    const token = await createAuthTokenForUser(testAdminUser.id, AllScopes)
+    apollo = await testApolloServer({
+      context: createTestContext({
+        auth: true,
+        userId: testAdminUser.id,
+        token,
+        role: testAdminUser.role,
+        scopes: AllScopes
+      })
+    })
+
+    const testWorkspace: BasicTestWorkspace = {
+      id: '',
+      ownerId: testAdminUser.id,
+      name: 'My Test Workspace'
+    }
+
+    await createTestWorkspace(testWorkspace, testAdminUser)
+
+    workspace = testWorkspace
```

Comment:
Why not just have `testWorkspace` be a const defined outside of `before()` next to `testAdminUser` and `testMemberUser`? This variable is redundant.

If you're worried about id being left empty, the `createTestWorkspace` call will fill it with the final ID. `ownerId` can be left as an empty string too, cause its gonna fill that based on the user you pass in too.